### DISCUSSION TITLE
Moving fuel consumption into a unified API

### DISCUSSION THREAD

**Discussion by wizardofgcc** [jeremyshannon/Exile#172] [Ideas]
I was looking into the code for lamps and fires recently, and I noticed the code for fuel consumption/burning was also quite similar. I think it could be moved into nodecrafting or minimal. To accommodate fires smoldering/turning to charcoal capability for custom checks during the timer could be added via callback functions.

**Comment by jeremyshannon** [comment]
Sounds good, I'll add it to my nodecrafting project sheet.
